src/actions/mod.rs

```diff
@@ -194,18 +194,54 @@ impl InitActionContext {
         struct BuildNotifier<O: Output> {
             out: O,
             active_build_count: Arc<AtomicUsize>,
+            progress_params: ProgressParams,
         }
 
+        // base progress parameters for notification of this build.
+        let progress_params = {
+            lazy_static! {
+                static ref ID_COUNTER: AtomicUsize = {
+                    AtomicUsize::new(0)
+                };
+            }
+
+            ProgressParams {
+                id: format!("progress_{}", ID_COUNTER.fetch_add(1, Ordering::SeqCst)),
```

Comment:
Is this ID format enough? Or do we need it to be globally unique, maybe introduce `uuid`?

Reply:
How are the ids designed to be used?

Reply:
As far as I understand they're just to tie together multiple progress notices in one long chain – they have the same ID. Whether or not that ID needs to be globally unique I'm not sure.